sdk/samples/iot/pnp/pnp_protocol.h

```diff
@@ -13,14 +13,22 @@
 #include <azure/core/az_span.h>
 #include <azure/iot/az_iot_hub_client.h>
 
+typedef enum
+{
+  PNP_METHODS_PUBLISH_TOPIC,
+  PNP_TELEMETRY_PUBLISH_TOPIC,
+  PNP_TWIN_DOCUMENT_PUBLISH_TOPIC,
+  PNP_TWIN_PATCH_PUBLISH_TOPIC
+} pnp_publish_topic;
+
 /**
- * @brief Callback which is called for each property found by the #pnp_process_twin_data()
- * API.
+ * @brief Callback which is invoked for each property found by the
+ * #pnp_process_device_twin_message() API.
  */
 typedef void (*pnp_property_callback)(
     az_span component_name,
-    const az_json_token* property_name,
-    az_json_reader property_value_as_json,
+    az_json_token const* property_name,
+    az_json_reader* property_value_as_json,
```

Comment:
this should stay as a value not pointer. Reason being we want to process the document on our end in a predictable manner and know the state. If the user somehow shifts the token to a place we don't expect, it will cause issues inside the function which invoked this callback.

Reply:
Do you mean shift the location of the token, or shift the value of the token?  We can keep the location in check by using az_json_reader *const, but def can't do az_json_reader const* to protect the token value because it will eventually be cast to void* in pnp_build_reported_property_with_status().